src/views/ActivitiesAll/index.js

```diff
@@ -58,10 +59,35 @@ export default class GordonActivitiesAll extends Component {
         activities,
         allActivities: activities,
         myInvolvements: myInvolvements,
-        loading: false,
+        loading: true,
         sessions,
         types,
       });
+
+      if (activities.length === 0) {
+        var recentSession;
+        recentSession = this.state.sessions[0].SessionCode;
```

Comment:
Do we need sessions from the state? Can't we use the variable instantiated within this method?